test/e2e/test/e2e.test.ts

```diff
@@ -131,6 +132,33 @@ describe('End to end tests for common events', () => {
     expect(sentryEvent.eventID).toMatch(eventId);
   });
 
+  // Currently doesn't work on Android because capturing screenshots doesn't work on emulator in GH Actions.
+  if (Platform.OS !== 'android') {
+    test('captureErrorReplay', async () => {
+      const element = await getElement('captureException');
+      await element.click();
+
+      const eventId = await waitForEventId();
+      const sentryEvent = await fetchEvent(eventId);
+      expect(sentryEvent.eventID).toMatch(eventId);
+
+      expect(sentryEvent.contexts).toBeDefined();
+      const replay = sentryEvent.contexts!['replay'] as any;
+      expect(replay).toBeDefined();
+      expect(replay.replay_id.length).toBe(32);
```

Comment:
Suggestion: We could use a more specific test to make it clear that we are dealing with an UUID instead of just checking for a fixed number.

```suggestion
const uuidRegex = /^[0-9a-f]{8}[0-9a-f]{4}[4][0-9a-f]{3}[89ab][0-9a-f]{3}[0-9a-f]{12}$/i;
expect(uuidRegex.test(replay.replay_id)).toBe(true);
```

Reply:
Thanks for the suggestion but I think it's an overkill here. The replay ID is read from the sentry server API so should already be a valid UUID. Additionally, it's used to fetch from other APIs by this ID so those calls would faile. It's not unlike all the other tests where we use an error ID to fetch the event from the API - there are no checks prior to API requests either.

Reply:
Oh I see, sound good to me